tests/test_cli.py

```diff
@@ -15,162 +17,438 @@ def cli_runner():
+            assert not result.exception
+            assert result.exit_code == 0
+
+    def test_update_outdated(self, cli_runner, tmpdir):
+        p = Project(tmpdir.strpath)
+        with chdir(p.path):
+            cli_runner.invoke(cli.install)
+            assert p.exists('.git/hooks/pre-commit')
+
+            hook = p.read('.git/hooks/pre-commit')
+            hook_hash = cli.get_hook_hash(p.abspath('.git/hooks/pre-commit'))
+
+            p.write('.git/hooks/pre-commit', hook.replace('# THERAPIST {}'.format(hook_hash), '# THERAPIST n0tth3h45h'))
 
             result = cli_runner.invoke(cli.install, input='y')
+            assert 'You are not using the current version of the pre-commit hook.' in result.output
             assert not result.exception
             assert result.exit_code == 0
 
+            hook = p.read('.git/hooks/pre-commit')
+            assert '# THERAPIST {}'.format(hook_hash) in hook
+
+    def test_dont_update_outdated(self, cli_runner, tmpdir):
+        p = Project(tmpdir.strpath)
+        with chdir(p.path):
+            cli_runner.invoke(cli.install)
+            assert p.exists('.git/hooks/pre-commit')
+
+            hook = p.read('.git/hooks/pre-commit')
+            hook_hash = cli.get_hook_hash(p.abspath('.git/hooks/pre-commit'))
+
+            p.write('.git/hooks/pre-commit', hook.replace('# THERAPIST {}'.format(hook_hash), '# THERAPIST n0tth3h45h'))
+
             result = cli_runner.invoke(cli.install, input='n')
+            assert 'Installation aborted.' in result.output
             assert result.exception
             assert result.exit_code == 1
 
-    def test_cli_install_outside_repo(self, cli_runner, tmpdir):
-        with chdir(tmpdir.strpath):
-            result = cli_runner.invoke(cli.install)
-        assert result.exception
-        assert result.exit_code == 1
+            hook = p.read('.git/hooks/pre-commit')
+            assert '# THERAPIST n0tth3h45h'.format(hook_hash) in hook
+
+    def test_preserve_existing(self, cli_runner, tmpdir):
+        p = Project(tmpdir.strpath)
+        with chdir(p.path):
+            p.write('.git/hooks/pre-commit', '')
+            hook = p.read('.git/hooks/pre-commit')
+            assert hook == ''
+
+            result = cli_runner.invoke(cli.install, input='y')
+            assert 'There is an existing pre-commit hook.' in result.output
+            assert 'Copying `pre-commit` to `pre-commit.legacy`...' in result.output
+            assert not result.exception
+            assert result.exit_code == 0
+
+            hook = p.read('.git/hooks/pre-commit')
+            assert '# THERAPIST' in hook
+
+            assert p.exists('.git/hooks/pre-commit.legacy')
+
+    def test_replace_existing(self, cli_runner, tmpdir):
+        p = Project(tmpdir.strpath)
+        with chdir(p.path):
+            p.write('.git/hooks/pre-commit', '')
+            hook = p.read('.git/hooks/pre-commit')
+            assert hook == ''
```

Comment:
Do you not trust `p.write()`? Maybe there should be a test for that instead of this sort of assertion in every test?